### PR TITLE
10704 better odata types

### PR DESCRIPTION
**Pull request by cooperka**
### Changes

⚠️ May be easier to view individual commits

- Improved types:
  - Authenticated media URLs
  - Geographic Hash
  - Cascading select Hash
  - Select multiple array
- Better "warner" info for OptionSets in use, because I needed to find one for testing and this seemed useful!

### CODE REVIEW

File: spec/fixtures/response_json/basic.json [hunk 1]
```diff
@@ -6,19 +6,30 @@
...
-  "SelectMultipleQ10": "[\"Cat\", \"Dog\"]",
+  "SelectMultipleQ10": [
+    "Cat",
+    "Dog"
+  ],
   "DatetimeQ11": "2015-10-12T12:15:12-06:00",
   "DateQ12": "2014-11-09",
   "TimeQ13": "23:15:00",
-  "ImageQ14": null
+  "ImageQ14": "__NEMO_HOST__/en/m/mission1/media/images/*image_id1*"
```
Comment (smoyth) [inline spec/fixtures/response_json/basic.json:34]: 🤩

File: lib/tasks/db_create_fake_data.rake [hunk 2]
```diff
@@ -55,9 +55,9 @@ namespace :db do
     end
 
     # Define media paths
-    image_path = Rails.root.join("spec", "fixtures", "media", "images", "the_swing.png").to_s
-    audio_path = Rails.root.join("spec", "fixtures", "media", "audio", "powerup.mp3").to_s
-    video_path = Rails.root.join("spec", "fixtures", "media", "video", "jupiter.mp4").to_s
+    image_path = Rails.root.join("spec/fixtures/media/images/the_swing.png")
+    audio_path = Rails.root.join("spec/fixtures/media/audio/powerup.mp3")
+    video_path = Rails.root.join("spec/fixtures/media/video/jupiter.mp4")
```
[the remark this answers is not in the document]
Comment (cooperka) [reply lib/tasks/db_create_fake_data.rake:60]: The default must have changed in a newer version of Rubocop (it was recently upgraded for NEMO). We don't explicitly specify anything in our config.

I can't immediately find the changelog, but see e.g. https://github.com/rubocop-hq/rubocop-rails/issues/195 for reasoning. Are you good with the new default?

File: app/models/results/response_json_generator.rb [hunk 3]
```diff
@@ -91,23 +93,42 @@ def answer_set_value(answer_set)
...
+        type: Media::ObjectsController.media_type(media.type),
+        mission_name: get_mission.compact_name,
+        locale: get_mission.default_locale
+      )
+      "#{BASE_URL_SIGNIFIER}#{path}"
+    end
+
+    def location_value(answer)
+      Answer::LOCATION_COLS.map do |key|
+        [key.titleize, answer.attributes[key]&.to_f]
```
[the remark this answers is not in the document]
Comment (cooperka) [reply app/models/results/response_json_generator.rb:128]: Hmm, they are returned as BigDecimal which gets rendered to JSON in scientific notation:

```
[123, answer.attributes[key]].to_s
=> [123, 0.15937378e2]
```

**Comment by cooperka** [comment]
Thanks Tom! Fixed "placeholder" naming, agree that's much better. Will merge after CI passes.

**Pull request merged**